ayon_bin_distro/gui/progress_ui.py

```diff
@@ -86,11 +107,23 @@ def update_progress(self):
 
 
 class MultiProgressWidget(QtWidgets.QWidget):
-    def __init__(self, controller, parent, delet_progress_bar_on_finish: bool = True):
+    def __init__(
+        self,
+        controller: worker.Controller,
+        parent: QtWidgets.QDialog,
+        delet_progress_bar_on_finish: bool = True,
+    ):
+        """core widget that holds and updates multiple Progress bars
+
+        Args:
+            controller: work.Controller instance that can be used to request the progress status
+            parent: the parent QDialog
+            delet_progress_bar_on_finish: bool decides if an progress bar will be deleted when if finished or not
+        """
         super().__init__(parent)
 
-        self._controller = controller
-        self._progress_bar_by_id = {}
+        self._controller: worker.Controller = controller
+        self._progress_bar_by_id: Dict[str, ProgressBar] = {}
 
         self._delet_progress_bar_on_finish: bool = delet_progress_bar_on_finish
```

Comment:
This may be easier to refactor in IDE
```suggestion
        self._delete_progress_bar_on_finish: bool = delete_progress_bar_on_finish
```